arcsi/view/item.py

```diff
@@ -1,16 +1,17 @@
 import requests
 
 from flask import current_app as app
-from flask import render_template, request, url_for
+from flask import render_template, url_for
 from flask_login import current_user
 from flask_security import roles_accepted
 
 from arcsi.view import router
 
 
 @router.route("/item/all")
+@roles_accepted("admin", "host", "guest")
 def list_items():
-    result = requests.get(app.config["APP_BASE_URL"] + url_for("arcsi.list_items"))
+    result = requests.get(app.config["APP_BASE_URL"] + url_for("arcsi.list_items"), headers = {"Authentication-Token": current_user.get_auth_token()})
```

Comment:
@tuz666 why does this work and other APIs don't when used internally? I thought that's the point of separation between login and auth. Or is there a difference if you call a GET/POST API or if you call it from a view? Sorry if the question is something we discussed already, I can't recall it now.

Reply:
Yes, there is a difference.
If you are logged in with a user it is possible to generate tokens (and add them to the headers) for GET requests which are used for view functions.
But you can't add headers to forms and that's why I had to create separate POST APIs for internal and external use.